File: bestia-messages/src/main/java/net/bestia/messages/LogoutMessage.java
package net.bestia.messages;

public class LogoutMessage extends Message {

	private static final long serialVersionUID = 1L;
	
	public static final String MESSAGE_ID = "system.logout";

	public LogoutMessage() {
		// no op.
	}
	
	public LogoutMessage(Message message) {
		setAccountId(message.getAccountId());
	}

	@Override
	public String getMessageId() {
		return MESSAGE_ID;
	}
}

File: bestia-messages/src/main/kotlin/net/bestia/messages/login/LogoutMessage.kt
package net.bestia.messages.login

/**
 * This message is send to the player in order to signal a (forced) logout from
 * the system.
 *
 * @author Thomas Felix
 */
data class LogoutMessage(
    val state: LoginResponse = LoginResponse.NO_REASON,
    val reason: String = ""
)